File: testproject/php/Engine.php
<?php
namespace testproject\php;

interface Store {}
class Base {}
class Settings {}
class Request {}

#[Route]
class Engine extends Base implements Store {
  public Settings $settings;
  
  #[Route]
  public function run(Request $req): Settings {
    require "config/bootstrap.php";
    return new Settings();
  }
}

File: testproject/scala/Engine.scala
package testproject.scala

import scala.io.Source

trait Store {}
class Base {}
class Settings {}
class Request {}

@deprecated
class Engine extends Base with Store {
  val settings: Settings = new Settings()
  
  @deprecated
  def run(req: Request): Settings = {
    val cfg = Source.fromFile("config/app.json")
    new Settings()
  }
}
